job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -530,9 +532,44 @@ class WebApi(system: ActorSystem,
    *    GET /healthz              - return OK or error message
    */
   def healthzRoutes: Route = pathPrefix("healthz") {
+    import ContextSupervisor._
+    import JobManagerActor._
+    import JobInfoActor._
+    import JobDAOActor._
     //no authentication required
     get { ctx =>
-      ctx.complete("OK")
+      logger.info("Receiving healthz check request")
```

Comment:
Your change looks okay, the only issue I see is that the definition of health check varies. Some people might want to check DB + Spark + HDFS also as part of health check and others might just want to know if WebApi is okay.

I would suggest, to hide your implementation behind an interface and provide 2 implementations. One could be the old one and other can be your implementation and it can be configurable via config file. 
An example of making it configurable can be found [here](https://github.com/spark-jobserver/spark-jobserver/blob/d705751418d65fd75a8ba2f928bab8baf2c2725e/job-server/src/main/resources/application.conf#L37) and its usage can be seen [here](https://github.com/spark-jobserver/spark-jobserver/blob/d705751418d65fd75a8ba2f928bab8baf2c2725e/job-server/src/main/scala/spark/jobserver/JobServer.scala#L70) and [here](https://github.com/spark-jobserver/spark-jobserver/blob/d705751418d65fd75a8ba2f928bab8baf2c2725e/job-server/src/main/scala/spark/jobserver/JobServer.scala#L115)

Reply:
Please don't hesitate to ask if you get stuck somewhere. Thanks.

Reply:
@bsikander Please review and merge if the changes are good.
- Made the health check configurable. 
- By default this returns OK, health check will be executed only if it is explicitly configured in application.conf
- Test is failing currently but it is not relevant to this change.

Reply:
Addressed